components/arm/arm.go

```diff
@@ -220,17 +221,61 @@ func (r *reconfigurableArm) MoveToPosition(
 	worldState *referenceframe.WorldState,
 	extra map[string]interface{},
 ) error {
+	model := r.actual.ModelFrame()
+	joints, err := r.actual.JointPositions(ctx, nil)
+	if err != nil {
+		return err
+	}
+	// check that joint positions are not out of bounds
+	if _, err = model.Transform(model.InputFromProtobuf(joints)); err != nil {
+		return err
+	}
 	r.mu.RLock()
 	defer r.mu.RUnlock()
 	return r.actual.MoveToPosition(ctx, pose, worldState, extra)
 }
 
 func (r *reconfigurableArm) MoveToJointPositions(ctx context.Context, positionDegs *pb.JointPositions, extra map[string]interface{}) error {
+	// check that joint positions are not out of bounds
+	if err := r.checkDesiredJointPositions(ctx, positionDegs.Values); err != nil {
+		return err
+	}
 	r.mu.RLock()
 	defer r.mu.RUnlock()
 	return r.actual.MoveToJointPositions(ctx, positionDegs, extra)
 }
 
+// checkDesiredJointPositions validates that the desired joint positions either bring the joint back
+// in bounds or do not move the joint more out of bounds.
+func (r *reconfigurableArm) checkDesiredJointPositions(ctx context.Context, desiredJoints []float64) error {
```

Comment:
[major] In general we don't ever want to be modifying the reconfigurable arm itself, and this method along with the other functionality introduced should work on the generic `arm.Arm` type

Reply:
Strongly agree with this

Reply:
Agreed. Nick, take a look at the `Move()` and `Plan()` functions at the end of `arm.go`; there are arm-generic Plan and Move functions there, that each individual arm driver calls for MoveToPosition. We probably want something similar to that for joint movement.